client/components/thank-you-v2/index.tsx

```diff
@@ -0,0 +1,36 @@
+import { ConfettiAnimation } from '@automattic/components';
+import { TranslateResult } from 'i18n-calypso';
+import ThankYouFooter, { ThankYouDetailProps } from './footer';
+import ThankYouHeader from './header';
+import ThankYouUpsell, { ThankYouUpsellProps } from './upsell';
+
+import './style.scss';
+
+interface ThankYouV2Props {
+	title: string | TranslateResult;
+	subtitle: string | TranslateResult;
+	headerButtons?: React.ReactNode;
+	products?: React.ReactNode;
```

Comment:
I'm not sure... Although, restricting the type here can be problematic, since we ultimately, we may not be expecting a list of `<ThankYouProduct>`, instead a list of components built using `<ThankYouProduct>`. For example, the domain only congrats page supplies a list of `<ProductDomain>`: https://github.com/Automattic/wp-calypso/pull/86186/files#diff-ae65199047ebe7decaec753ef0d3314fde82a4fe31b8e011c417a3c4395a6cf1R75-R86 

We'll have to be specific in the documentations so devs know this requirements in order to avoid deviating from the standardized styles for thank you product tiles.

Reply:
That's mostly curiosity but I'd like to summon @nightnei to ask the Lord Of TypeScript if that's something doable.

Reply:
We can certainly look into this further. Although, I would suggest leaving this as-is for now to get this PR out of the way and unblock other relevant work.

Reply:
@stephanethomas nice question, honestly I didn't do it and haven't seen cases when it was necessary to do, so hard to answer. But theoretically, it's possible to do with the help of generic types.